### PR TITLE
Add Dorogovtsev-Goltsev-Mendes graph generator

### PR DESCRIPTION
**Pull request by dfacoet**
Related to #150.

This adds generator functions for Dorogovtsev-Goltsev-Mendes graphs to both rustworkx and rustworkx-core.

First contribution, feedback is welcome.

### CODE REVIEW

File: tests/generators/test_dorogovtsev_goltsev_mendes.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# under the License.
+
+import unittest
+
+import rustworkx
+
+
+class TestDorogovtsevGoltsevMendesGraph(unittest.TestCase):
+    def test_dorogovtsev_goltsev_mendes_graph(self):
+        for n in range(0, 6):
```
Comment (IvanIsCoding) [inline tests/generators/test_dorogovtsev_goltsev_mendes.py:20]: Use subtests like in https://github.com/Qiskit/rustworkx/blob/8af19f0f9c3f80463e39bb691b8f13e96a5d243f/tests/graph/test_subgraph_isomorphic.py#L282

File: tests/generators/test_dorogovtsev_goltsev_mendes.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import rustworkx
+
+
+class TestDorogovtsevGoltsevMendesGraph(unittest.TestCase):
```
Comment (IvanIsCoding) [inline tests/generators/test_dorogovtsev_goltsev_mendes.py:18]: You should add test cases for invalid inputs e.g. n = -1

File: tests/generators/test_dorogovtsev_goltsev_mendes.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    def test_dorogovtsev_goltsev_mendes_graph(self):
+        for n in range(0, 6):
+            graph = rustworkx.generators.dorogovtsev_goltsev_mendes_graph(n)
+            self.assertEqual(len(graph), (3**n + 3) // 2)
+            self.assertEqual(len(graph.edges()), 3**n)
```
Comment (IvanIsCoding) [inline tests/generators/test_dorogovtsev_goltsev_mendes.py:23]: I think it is also worth checking that the graph you generate is planar. We have https://www.rustworkx.org/apiref/rustworkx.is_planar.html

**Review by IvanIsCoding — COMMENTED** [review]
Overall Rust code looks good. I did leave some comments about the Python tests though. I am not an expert on this type of graph but I think we can add some code to ensure at least it keeps the code properties

File: tests/generators/test_dorogovtsev_goltsev_mendes.py (hunk 1 above)
Comment (dfacoet) [reply tests/generators/test_dorogovtsev_goltsev_mendes.py:20]: Should it be one subTest per assertion, or just one for each n?

Comment (IvanIsCoding) [reply tests/generators/test_dorogovtsev_goltsev_mendes.py:20]: Subtest for each N. The assertions go in the same subtest

**Comment by dfacoet** [comment]
Thanks @IvanIsCoding for the review, I made the changes to the tests.

**Review by IvanIsCoding — APPROVED** [review]
LGTM, thanks for addressing the feedback quickly! Hopefully we release it in 0.15 soon-ish

**Pull request merged**